climpred/comparisons.py

```diff
@@ -72,24 +71,39 @@ def _m2m(ds, supervector_dim='svd'):
     forecast_list = []
     for m in ds.member.values:
         # drop the member being reference
-        ds_reduced = _drop_members(ds, rmd_member=[m])
+        if stack:
+            ds_reduced = _drop_members(ds, rmd_member=[m])
+        else:
+            # TODO: when not stacked I have one identical
```

Comment:
?

Reply:
this TODO is partly resolved by this correction factor I introduced after perfect_model skill is computed. the problem with m2m is that I would get empty elements in the diagonal when building up the comparison. those empty all-nan arrays then make deterministic metrics crash. I dont like this status quo, but cannot find a better solution for now.

Reply:
So does this crash still happen? Or is there an error catch needed here? Or is that what this `else` is doing? Can you update the `TODO` to be more clear what the problem is?

Reply:
no crashes now. updated TODO comments.